public/app/features/correlations/CorrelationsPage.tsx

```diff
@@ -44,10 +45,30 @@ export default function CorrelationsPage() {
   const canWriteCorrelations = contextSrv.hasPermission(AccessControlAction.DataSourcesWrite);
 
   const handleAdd = useCallback(() => {
+    reportInteraction('grafana_correlations_added');
     fetchCorrelations();
     setIsAdding(false);
   }, [fetchCorrelations]);
 
+  const handleUpdate = useCallback(() => {
+    reportInteraction('grafana_correlations_edited');
+    fetchCorrelations();
+  }, [fetchCorrelations]);
+
+  const handleDelete = useCallback(
+    (params: RemoveCorrelationParams) => {
+      remove.execute(params);
+    },
+    [remove]
+  );
+
+  // onDelete - triggers when deleting a correlation
```

Comment:
What does 'onDelete` refer to?

Reply:
just an alias for the effect to make it clear it's triggered when deleting a correlation, unlike the `handleAdd` and `handleUpdate` (which i renamed to `handleAdded` and `handleUpdated` in ba002760a1 to be more descriptive) that are triggered only after a successful creation/edit, `handleDelete` is not handled internally by another child component and it's called every time the delete button is clicked. this effect is to handle only successful deletions (so that events are triggered consistently only when the relative operation is successful)

Reply:
> (...) triggered consistently only when the relative operation is successful

If so, should it be:

```suggestion
  // onDeleted - triggers after deleting a correlation
```

It could also be described with more verbose code instead of comments, e.g.

```typescript
  const handleDeleted = () => {
    reportInteraction('grafana_correlations_deleted');
  }
  
  useEffect(() => {
    if (remove.value) {
      handleDeleted();
    }
  }, [remove.value]);
```

Just a personal preference, so feel free to ignore it.

Reply:
> so that events are triggered consistently only when the relative operation is successful

by events in that sentence i meant "interactions". a bit confusing given i also used `event` to refer to the event handler before, sorry 🙏 

to recap: `handleDelete` is triggered before the deletion happen, the effect makes it so the interaction is only reported after a successful deletion. hence why it is called `handleDelete` and not `deleted` like the others 2.


regarding
> It could also be described with more verbose code instead of comments

that code would run on every render